Relocate internal Compass docs and implementation details (#1124)
kyma-incubator/compass: docs/compass/04-01-installation.md

# Enable Compass in Kyma

To enable Compass in Kyma, follow the instructions for the [custom component installation](https://kyma-project.io/docs/master/root/kyma#configuration-custom-component-installation) and enable the `compass` and `compass-runtime-agent` modules. You can also [install Kyma on a cluster](https://kyma-project.io/docs/master/root/kyma#installation-install-kyma-on-a-cluster) with the ready-to-use configurations for different modes. There are two modes in which you can enable Compass in Kyma: default Kyma installation and Compass as a Central Management Plane.

## Default Kyma installation

![Kyma mode1](./assets/kyma-mode1.svg)

> This is a single-tenant mode which provides the complete cluster Kyma installation with all components, including Compass and the Runtime Agent. In this mode, Runtime Agent is already connected to Compass and they both work in a single-tenant mode as well. Using this mode, you can register external Applications in Kyma. To enable it, follow the cluster Kyma installation using the [`installer-cr-cluster-with-compass.yaml.tpl`](https://github.com/kyma-project/kyma/blob/master/installation/resources/installer-cr-cluster-with-compass.yaml.tpl) configuration file. Before you start the installation, apply the following ConfigMap which enables the [API Packages](https://github.com/kyma-incubator/compass/blob/master/docs/compass/03-packages-api.md) functionality:

Reviewer: ```suggestion
This is a single-tenant mode which provides the complete cluster Kyma installation with all components, including Compass and the Runtime Agent. In this mode, the Runtime Agent is already connected to Compass. Using this mode, you can register external Applications in Kyma. To enable it, follow the cluster Kyma installation using the [`installer-cr-cluster-with-compass.yaml.tpl`](https://github.com/kyma-project/kyma/blob/master/installation/resources/installer-cr-cluster-with-compass.yaml.tpl) configuration file. The default [legacy mode](#architecture-application-connector-components-application-operator) used in Kyma does not support integration with Compass. For that reason, before you start the installation, apply the following ConfigMap which disables components used in the legacy mode, such as the Application Registry and the Connector Service: 
```
This is outdated. See [this](https://github.com/kyma-project/kyma/pull/7876/files) PR.

```yaml
cat <<EOF | kubectl apply -f -
apiVersion: v1
kind: ConfigMap
metadata:

>   name: api-packages-override

Reviewer: ```suggestion
  name: disable-legacy-connectivity-override
```
This is outdated. The flag name has changed. See [this](https://github.com/kyma-project/kyma/pull/7876/files) PR.

  namespace: kyma-installer
  labels:
    installer: overrides
    kyma-project.io/installation: ""
data:

>   global.enableAPIPackages: "true"

Reviewer: ```suggestion
  global.disableLegacyConnectivity: "true"
```
This is outdated. The flag name has changed. See [this](https://github.com/kyma-project/kyma/pull/7876/files) PR.

 [`installer-cr-cluster-compass.yaml.tpl`](https://github.com/kyma-project/kyma/blob/master/installation/resources/installer-cr-cluster-compass.yaml.tpl) configuration file:

```yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: compass-overrides
  namespace: kyma-installer
  labels:
    installer: overrides
    component: compass
    kyma-project.io/installation: ""
data:
  # The parameter that enables the Compass gateway, as the default Kyma gateway is disabled in this installation mode
  gateway.gateway.enabled: "true"
  # The name of the currently used gateway
  global.istio.gateway.name: compass-istio-gateway
  # The Namespace of the currently used gateway
  global.istio.gateway.namespace: compass-system
  # The parameter that disables preconfiguration for the Compass Runtime Agent
  global.agentPreconfiguration: "false"
  # The Namespace with a Secret that contains a certificate for the Connector Service
  global.connector.secrets.ca.namespace: compass-system
```

### Kyma Runtime

> This is a single-tenant mode that provides complete cluster Kyma installation with the Runtime Agent. To enable this mode, follow the cluster Kyma installation using the [`installer-cr-cluster-runtime.yaml.tpl`](https://github.com/kyma-project/kyma/blob/master/installation/resources/installer-cr-cluster-runtime.yaml.tpl) configuration file. Before you start the installation, apply the following ConfigMap which enables the [API Packages](https://github.com/kyma-incubator/compass/blob/master/docs/compass/03-packages-api.md) functionality:

Reviewer: ```suggestion
This is a single-tenant mode that provides complete cluster Kyma installation with the Runtime Agent. To enable this mode, follow the cluster Kyma installation using the [`installer-cr-cluster-runtime.yaml.tpl`](https://github.com/kyma-project/kyma/blob/master/installation/resources/installer-cr-cluster-runtime.yaml.tpl) configuration file. The default [legacy mode](#architecture-application-connector-components-application-operator) used in Kyma does not support integration with Compass. For that reason, before you start the installation, apply the following ConfigMap which disables components used in the legacy mode, such as the Application Registry and the Connector Service:
```

```yaml
cat <<EOF | kubectl apply -f -
apiVersion: v1
kind: ConfigMap
metadata:

>   name: api-packages-override

Reviewer: ```suggestion
  name: disable-legacy-connectivity-override
```
This is outdated. The flag name has changed. See [this](https://github.com/kyma-project/kyma/pull/7876/files) PR.

  namespace: kyma-installer
  labels:
    installer: overrides
    kyma-project.io/installation: ""
data:

>   global.enableAPIPackages: "true"

Reviewer: ```suggestion
  global.disableLegacyConnectivity: "true"
```
This is outdated. The flag name has changed. See [this](https://github.com/kyma-project/kyma/pull/7876/files) PR.
